docs/data/data-grid/pivoting/pivoting.md

```diff
@@ -2,16 +2,170 @@
+For example, if you pivot with **Region** and **Quarter** for columns, each unique combination of the two will become a column in the pivot Grid.
+Try selecting the **Quarter** checkbox in the **Columns** section in the demo above to see this.
+
+### Pivoting in action
+
+In the demo below, the goal is to evaluate sales by region and quarter for each product and size available.
+To accomplish this, the dataset can be pivoted with **Product** and **Size** for rows; **Region** and **Quarter** for columns; and **Sales** and **Profit** for aggregated values.
+
+{{"demo": "GridPivotingMultipleValues.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Implementing pivoting
+
+### Pivot model
+
+The pivot model is a configuration object that defines rows, columns, and values of the pivot Grid:
+
+```tsx
+interface GridPivotModel {
+  rows: Array<{
+    field: string;
+    hidden?: boolean;
+  }>;
+  columns: Array<{
+    field: string;
+    sort?: 'asc' | 'desc';
+    hidden?: boolean;
+  }>;
+  values: Array<{
+    field: string;
+    aggFunc: string;
+    hidden?: boolean;
+  }>;
+}
+```
+
+### Initialize pivoting
+
+Use the `initialState` prop to initialize uncontrolled pivoting.
+This is the recommended method unless you specifically need to control the state.
+You can initialize the pivot model, toggle pivot panel visibility, and toggle the pivot mode as shown below:
+
+```tsx
+<DataGridPremium
+  initialState={{ pivoting: { model: pivotModel, active: true, panelOpen: true } }}
+/>
+```
+
+{{"demo": "GridPivotingInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Controlled pivoting
+
+For fully controlled pivoting state, you can use the following props:
+
+- Pivot model:
+  - `pivotModel`: Controls the current pivot configuration.
+  - `onPivotModelChange`: Callback fired when the pivot model changes.
+- Pivot mode toggle:
+  - `pivotActive`: Controls whether pivot mode is active.
+  - `onPivotActiveChange`: Callback fired when the pivot mode changes between active and inactive.
+- Pivot panel:
+  - `pivotPanelOpen`: Controls whether the pivot panel is open.
+  - `onPivotPanelOpenChange`: Callback fired when the pivot panel is opened or closed.
+
+{{"demo": "GridPivotingControlled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Using fields in the pivot model multiple times
+
+While this is not supported yet, we are working to bring this feature to the Data Grid.
+
+## Disable pivoting
+
+To disable pivoting, set the `disablePivoting` prop to `true`:
+
+```tsx
+<DataGridPremium disablePivoting />
+```
+
+## Derived columns in pivot mode
+
+In pivot mode, it's often useful to group data by a year or quarter.
+The Data Grid automatically generates year and quarter columns for each **Date** column for this purpose.
+
+In the demo below, the **Transaction Date** column is represented by additional columns in pivot mode: **Transaction Date (Year)** and **Transaction Date (Quarter)**:
```

Comment:
I think we should keep using the example from sales you started the page with. 

It's a good example of the pain point, now that the flat dataset doesn't come with Quarter column.

```suggestion
For example, the sales dataset we've been using throughout the examples has a "Quarter" column, but often in real applications, each sale would have a precise Transaction date, like in the following demo.

The **Transaction Date** column is represented by additional columns in pivot mode: **Transaction Date (Year)** and **Transaction Date (Quarter)**:
```